### ISSUE TITLE
Code-review

### ISSUE THREAD

**Issue by KudzaiDhewa** [KudzaiDhewa/Stat212_Project1#4]
**Research Questions Summary**

The big picture of our project is to figure out ways to maximize a player’s survival chances in PUBG. If a player’s team is the last team standing, we will be referring to them as the chicken dinner winner. In total we have 3 research questions that revolve around exploring tactics skillful players apply to survive the longest, or to effectively participate in combat to directly eliminate other teams to secure the top spots. Specifically, we aim to explore data retrieved from 3 important endpoints of the PUBG API: lifetime stats, weapon master and survival mastery. We anticipate that by looking at metrics such as killCount, KnockCount, weaponUsed, modeOfTransport, etc, we will be able to come up with some visualizations that effectively demonstrate strong correlations between a skill set and survival probability.

**Planned Digital Artifact**

Due to the nature of our data, which is a survival game, we agree that a blog would be our best option for our digital artifact. The rationale behind this decision is that, compared to other data such as the census data, where variable interpretability can be easy and straightforward and any user can freely pick and choose filters in a shiny app, data specific to a game can be a little harder to interpret, unless you are a gaming enthusiast. Therefore, a blogpost written by expert players, like we are, will be most effective in conveying the most interesting observations and in explaining why such observations are worth paying attention to.

In general, we will be working on a blog that tells a story about PUBG players’ performances through the analysis of their lifetime statistics, weapon mastery and survival mastery. This will be done through a variety of visualizations and descriptive text mainly focusing on key takeaways that could help both players and game developers enhance the gaming experience for PUBG and similar games!

Group: Joshua/Shelley/Sydney

**Comment by Joshua-Segebre** [comment]
**A brief description of the code you reviewed** 
- I reviewed the weapon_mastery.qmd code. Which analyzes different types of weapons in a popular video game and their statistics in order to improve player survival outcomes and better the game overall. 

**A list of things you think the author did well**
- I think the authors did a GREAT job at including comments and their code is organized into specific sections that make it easy to refer to. Despite the code complexity, it was easy to understand what to code is doing.

**A list of things that you think the author could improve** 
- I think adding a visualization could add to the overall argument. It would also be interesting to see if there are any missing data on players and if that has anything to do with the weapon type

**A list of questions you have about the code**
- Will the code be joined together? What types of relationships do you want to be represented from the datasets to your audience?

**Comment by shellybeee** [comment]
**Code Review** 
Review of weapon_mastery.qmd
About: Code document including API code for weapon master data
- Great code separation and use of headers to indicate different sections of your code
- I like that you are implementing pseudocode for things that are in progress; I think you should still keep short comments in the final code (either above or beside the code).
- Do replace API with key 

lifetimestats.qmd
API getting code for Lifetimestats info.
Takes in players account ID and API key to create a dataframe for one player. 
Includes a test with a single account ID
- good use of functions and test cases
- Use more comments that go step-by-step
-

**Comment by sydneyohr** [comment]
- We reviewed the weapons_mastery.qmd that is pulling data from the PUBG API and extracting weapon mastery of players. So far they have a sample function that uses one player's account and pulls information about that player and they are working on generalizing that function so it can work on multiple players. 
- Hanna did a great job with pseudocode and labeling what each section does. It is very neat and organized, which made it easy decipher the code. 
- One thing I would do is to remove the API key from the qmd
- I think it would be cool to add a blurb about the final viz

**Comment by Tam-Ngn** [comment]
> **A brief description of the code you reviewed**
> 
> * I reviewed the weapon_mastery.qmd code. Which analyzes different types of weapons in a popular video game and their statistics in order to improve player survival outcomes and better the game overall.
> 
> **A list of things you think the author did well**
> 
> * I think the authors did a GREAT job at including comments and their code is organized into specific sections that make it easy to refer to. Despite the code complexity, it was easy to understand what to code is doing.
> 
> **A list of things that you think the author could improve**
> 
> * I think adding a visualization could add to the overall argument. It would also be interesting to see if there are any missing data on players and if that has anything to do with the weapon type
> 
> **A list of questions you have about the code**
> 
> * Will the code be joined together? What types of relationships do you want to be represented from the datasets to your audience?

Yes, the right visualizations are our most important next step. In the end, we plan on synthesizing all viz from the 3 endpoints so we can end up with a reasonable argument on what a player should do to maximize their chance of winning a match, or just the chance of survival in general.

**Comment by Tam-Ngn** [comment]
> **Code Review** Review of weapon_mastery.qmd About: Working code document including API code for weapon master data
> 
> * Great code separation and use of headers to indicate different sections of your code
> * I like that you are implementing pseudocode for things that are in progress; I think you should still keep short comments in the final code (either above or beside the code).
> * Do replace API key with an API text file for your final code
> 
> Review of lifetimestats.qmd About: Includes API code for Lifetimestats info. Function takes in players account ID and API key to create a dataframe for one player. Includes a test with a single account ID
> 
> * Good use of functions and test cases!
> * Use more comments that outline step-by-step progression for review.
> * For complex functions, Include a brief (1-sentence) outline of what the function aims to do and returns

We will put our keys in a separate private folder and use gitignore to hide them when pushing the folder to github

**Comment by KudzaiDhewa** [comment]
Thank you for your feedback. Yes, we are working on putting together the right visualizations. As a result of your comment on joining datasets we have also come up with a plan to weave the lifetime stats, weapon and survival mastery data to tell a single story about how PUGB players can improve from beginner to elite

**Comment by KudzaiDhewa** [comment]
> * We reviewed the weapons_mastery.qmd that is pulling data from the PUBG API and extracting weapon mastery of players. So far they have a sample function that uses one player's account and pulls information about that player and they are working on generalizing that function so it can work on multiple players.
> * Hanna did a great job with pseudocode and labeling what each section does. It is very neat and organized, which made it easy decipher the code.
> * One thing I would do is to remove the API key from the qmd
> * I think it would be cool to add a blurb about the final viz

Thanks Sydney, we will definitely hide our API keys
